src/loader/LoaderUtils.ts

```diff
@@ -0,0 +1,202 @@
+import { 
+  AssetPromise, 
+  Engine, 
+  request, 
+  Texture2D, 
+  AssetType,
+  TextureFilterMode,
+  TextureWrapMode,
+} from "@galacean/engine";
+import { 
+  TextureAtlas, 
+  AtlasAttachmentLoader,
+  SkeletonJson,
+  SkeletonBinary,
+  SkeletonData,
+  Texture,
+  TextureFilter,
+  TextureWrap,
+} from "@esotericsoftware/spine-core";
+import { SpineAssetBundle } from "./SpineLoader";
+
+export async function loadAndCreateSpineSkeletonData(
+  bundle: SpineAssetBundle,
+  engine: Engine,
+): Promise<SkeletonData> {
+  const { skeletonPath, atlasPath, imagePaths, skeletonExtension, imageExtensions } = bundle;
+  const skeletonLoadFunc = skeletonExtension === 'json' ? loadText : loadBinary;
+  let skeletonTextData: string | ArrayBuffer;
+  let textureAtlas: TextureAtlas;
+  if (imagePaths.length > 0) {
+    let atlasText: string;
+    let textures: Texture2D[];
+    const loadQueue: AssetPromise<any>[] = [
+      skeletonLoadFunc(skeletonPath),
+      loadText(atlasPath),
+    ];
+    const texturePromises: AssetPromise<any>[] = imagePaths.map((imagePath, index) => {
+      const ext = imageExtensions[index];
+      let imageLoaderType = AssetType.Texture2D;
+      if (ext === 'ktx') imageLoaderType = AssetType.KTX;
+      if (ext === 'ktx2') imageLoaderType = AssetType.KTX2;
+      return loadTexture(imagePath, engine, imageLoaderType);
+    });
+    loadQueue.push(AssetPromise.all(texturePromises));
+    try {
+      [skeletonTextData, atlasText, textures] = await Promise.all(loadQueue);
+      textureAtlas = new TextureAtlas(atlasText);
+      let i = 0;
+      for (let page of textureAtlas.pages) {
+        if (textures[i]) {
+          const texture = createAdaptiveTexture(textures[i]);
+          page.setTexture(texture);
+          i++;
+        }
+      }
+    } catch (error) {
+      throw error;
+    }
+  } else {
+    try {
+      [skeletonTextData, textureAtlas] = await Promise.all([
+        skeletonLoadFunc(skeletonPath),
+        loadTextureAtlas(atlasPath, engine),
+      ]);
+    } catch(error) {
+      throw error;
+    }
+  }
+  const atlasLoader = new AtlasAttachmentLoader(textureAtlas);
+  if (skeletonExtension === 'json') {
+    return new SkeletonJson(atlasLoader).readSkeletonData(skeletonTextData);
+  } else {
+    return new SkeletonBinary(atlasLoader).readSkeletonData(new Uint8Array(skeletonTextData as ArrayBuffer));
+  }
+}
+
+function loadText(url: string): AssetPromise<string> {
+  return request(url, { type: "text" });
+}
+
+function loadBinary(url: string): AssetPromise<ArrayBuffer> {
+  return request(url, { type: "arraybuffer" });
+}
```

Comment:
why not use BufferLoader

Reply:
BufferLoader can only load .txt file. It not support spine .json and .atlas files.